documentation/asciidoc/computers/compute-module/cmio-camera.adoc

```diff
@@ -1,118 +1,331 @@
+----
+
+
+=== I2C mapping of GPIO pins
+
+By default, the supplied camera drivers assume that CAM1 uses `i2c-10` and CAM0 uses `i2c-0`. Compute module I/O boards map the following GPIO pins to `i2c-10` and `i2c-0`:
+
+[%header,cols="1,1,1"]
+|===
+| I/O Board Model
+| `i2c-10` pins
+| `i2c-0` pins
+
+| CM4 I/O Board
+| GPIOs 44,45
+| GPIOs 0,1
+
+| CM1, CM3, CM3+, CM4S I/O Board
+| GPIOs 0,1
+| GPIOs 28,29
+|===
+
+To connect a camera to the CM1, CM3, CM3+ and CM4S I/O Board, add the following directive to `/boot/firmware/config.txt` to accommodate the swapped pin assignment:
+
+----
+dtoverlay=cm-swap-i2c0
+----
+
+Alternative boards may use other pin assignments. Check the documentation for your board and use the following alternate overrides depending on your layout:
+
+[%header,cols="1,1"]
+|===
+| Swap
+| Override
+
+| Use GPIOs 0,1 for i2c0
+| `i2c0-gpio0`
+
+| Use GPIOs 28,29 for i2c0 (default)
+| `i2c0-gpio28`
+
+| Use GPIOs 44&45 for i2c0
+| `i2c0-gpio44`
+
+| Use GPIOs 0&1 for i2c10 (default)
+| `i2c10-gpio0`
+
+| Use GPIOs 28&29 for i2c10
+| `i2c10-gpio28`
+
+| Use GPIOs 44&45 for i2c10
+| `i2c10-gpio44`
+|===
+
+==== GPIO pins for shutdown
+
+For camera shutdown, Device Tree assumes `cam1_reg` and `cam0_reg`.
+
+The CM4 IO Board provides a single GPIO pin for both aliases, so both cameras share the same regulator.
+
+The CM1, CM3, CM3+, and CM4S I/O Board provides no GPIO pin for `cam1_reg` and `cam0_reg`, so the regulators are disabled on those boards. However, you can enable them with the following directives in `/boot/firmware/config.txt`:
+
+* `dtparam=cam1_reg`
+* `dtparam=cam0_reg`
+
+To assign `cam1_reg` and `cam0_reg` to a specific pin on a custom board, use the following directives in `/boot/firmware/config.txt`:
+
+* `dtparam=cam1_reg_gpio<pin number>`
+* `dtparam=cam0_reg_gpio<pin number>`
+
+For example, to use pin 42 as the regulator for CAM1, add the directive `dtparam=cam1_reg_gpio42` to `/boot/firmware/config.txt`.
+
+These directives only work for GPIO pins connected directly to the SoC, not for expander GPIO pins.
+
+If a custom board doesn't work with these overlays, create a new Device Tree that describes the entire board.
+
+==== Configure camera pins with Device Tree
+
+By default, the GPIOs used for the camera on the Compute Module are set to input mode. To xref:configuration.adoc#changing-the-default-pin-configuration[override these default settings], create a custom xref:configuration.adoc#device-trees-overlays-and-parameters[Linux Device Tree]. To define a custom Device Tree, create a custom a Device Tree Source (`dts`) file. To load your custom Device Tree, xref:configuration.adoc#changing-the-default-pin-configuration[compile your Device Tree Source file into a `dt-blob.bin` file]and place it in the boot partition. Raspberry Pi OS will load this Device Tree into firmware when the system boots.
+
+The following Device Tree Source files enable single and dual cameras for CM1, CM3, and CM4:
```

Comment:
Are these _example_ Device Tree Source files?
(and another minor nitpick, but I think this is the first place where you don't explicitly mention CM3+ or CM4S :wink: )

Reply:
The omission is intentional -- these Device Tree Source files don't include CM3+ or CM4S because they predate those Compute Modules. Sadly I don't have the bandwidth (or the material access) to build those entire source trees in this PR.